engine/schema/mysql.sql

```diff
@@ -3,41 +3,53 @@
 --
 
 -- record membership in an access collection
-CREATE TABLE `prefix_access_collection_membership` (
-  `user_guid` int(11) NOT NULL,
-  `access_collection_id` int(11) NOT NULL,
-  PRIMARY KEY (`user_guid`,`access_collection_id`)
-) ENGINE=MyISAM DEFAULT CHARSET=utf8;
+-- Access collections were replaced by ACLs
+-- See https://docs.google.com/a/elgg.org/document/d/1R3v_bYno6fw8mV5_GDW93uTI0pvR9S4yQ-BHMKTz1zs/edit#
+-- Each user-created "friend collection" is now an entity pointing to an ACL
 
 -- define an access collection
-CREATE TABLE `prefix_access_collections` (
-  `id` int(11) NOT NULL AUTO_INCREMENT,
-  `name` text NOT NULL,
-  `owner_guid` bigint(20) unsigned NOT NULL,
-  `site_guid` bigint(20) unsigned NOT NULL DEFAULT '0',
+-- Access collections were replaced by ACLs
+
+-- internal type strings. Think of this as metastrings for code.
+-- e.g. entity type, entity subtype, type of lists (the old relationship name)
+CREATE TABLE `prefix_types` (
+  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+  `type` varchar(255) NOT NULL DEFAULT '',
   PRIMARY KEY (`id`),
-  KEY `owner_guid` (`owner_guid`),
-  KEY `site_guid` (`site_guid`)
-) ENGINE=MyISAM AUTO_INCREMENT=3 DEFAULT CHARSET=utf8;
+  UNIQUE KEY `idx_type` (`type`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
 
 -- store an annotation on an entity
-CREATE TABLE `prefix_annotations` (
-  `id` int(11) NOT NULL AUTO_INCREMENT,
-  `entity_guid` bigint(20) unsigned NOT NULL,
-  `name_id` int(11) NOT NULL,
-  `value_id` int(11) NOT NULL,
-  `value_type` enum('integer','text') NOT NULL,
-  `owner_guid` bigint(20) unsigned NOT NULL,
-  `access_id` int(11) NOT NULL,
-  `time_created` int(11) NOT NULL,
-  `enabled` enum('yes','no') NOT NULL DEFAULT 'yes',
+-- Annotations are now entities with type=annotation and container=(the annotated entity)
+
+-- lists are ordered GUIDs identified by /GUID/type. e.g. /123/members
+CREATE TABLE `prefix_lists` (
+  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
+  `type_id` int(11) NOT NULL,
+  `target_guid` bigint(20) unsigned NOT NULL,
```

Comment:
As @jdalsem suggested, I think we could generate user-facing ACL names by passing target/type through a hook.